pkg/services/sso/redhatsso_service.go

```diff
@@ -290,6 +291,22 @@ func (r *redhatssoService) CreateServiceAccountInternal(accessToken string, requ
 }
 
 func (r *redhatssoService) DeleteServiceAccountInternal(accessToken string, clientId string) *errors.ServiceError {
+	if strings.HasPrefix(clientId, "canary") || strings.HasPrefix(clientId, "kas-fleetshard-agent") {
+		first := 0
+		max := 100
+		for {
```

Comment:
if the account is not found we might have an infinity loop here.
We might want to halt the process when the `accounts` in line number: 298 is an empty list.